shared/AppInsightsCore/src/JavaScriptSDK.Interfaces/IDiagnosticLogger.ts

```diff
@@ -33,7 +34,7 @@ export interface IDiagnosticLogger {
     /**
      * The internal logging queue
      */
-    queue: _InternalLogMessage[];
+    queue:ILogMessageQueue;
```

Comment:
As a public property this is a breaking change, while we don't expect people to be access it we can't make this change

Reply:
To implement the queue length checks / limits we will need to do this as a helper function to update the check and reduce the queue.

The only other "safe" way to do this would be to construct a new object that implements the COMPLETE array accessors and functions which from a space and performance perspective is not viable and we would also need to use "get" which is not compatible with our current ES3 limitation 😢